include/deal.II/base/data_out_base.h

```diff
@@ -2884,9 +2936,68 @@ class XDMFEntry
     &attribute_dims;
   }
 
-  /// Get the XDMF content associated with this entry.
-  /// If the entry is not valid, this returns an empty string.
+  /**
+   * Get the XDMF content associated with this entry.
+   * If the entry is not valid, this returns an empty string.
+   */
   std::string get_xdmf_content(const unsigned int indent_level) const;
+
+private:
+  /**
+   * Whether this entry is valid and contains data to be written.
+   */
+  const bool valid;
+
+  /**
+   * The name of the HDF5 heavy data solution file this entry references.
+   */
+  const std::string h5_sol_filename;
+
+  /**
+   * The name of the HDF5 mesh file this entry references.
+   */
+  const std::string h5_mesh_filename;
+
+  /**
+   * The simulation time associated with this entry.
+   */
+  const double entry_time;
+
+  /**
+   * The number of data nodes.
+   */
+  const unsigned int num_nodes;
+
+  /**
+   * The number of data cells.
+   */
+  const unsigned int num_cells;
+
+  /**
+   * The dimension associated with the data.
+   */
+  const unsigned int dimension;
+
+  /**
+   * The dimension of the space the data lives in.
+   * Note that dimension <= space_dimension.
+   */
+  const unsigned int space_dimension;
+
+  /**
+   * The attributes associated with this entry and their dimension.
+   */
+  std::map<std::string, unsigned int> attribute_dims;
+
+  /**
+   * Small function to create indentation for XML file.
+   */
+  std::string indent(const unsigned int indent_level) const
+  {
+    std::string res = "";
+    for (unsigned int i=0; i<indent_level; ++i) res += "  ";
+    return res;
+  }
```

Comment:
same here -- move into the .cc file.

Though this could also be a `static` function, and consequently could be a member of an anonymous namespace.